packages/components/src/segmented-control/component.tsx

```diff
@@ -0,0 +1,131 @@
+/**
+ * External dependencies
+ */
+import { cx } from 'emotion';
+// eslint-disable-next-line no-restricted-imports
+import { RadioGroup, useRadioState } from 'reakit';
+import useResizeAware from 'react-resize-aware';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useRef, useMemo } from '@wordpress/element';
+import { useMergeRefs } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import {
+	contextConnect,
+	useContextSystem,
+	PolymorphicComponentProps,
+} from '../ui/context';
+import { View } from '../view';
+import * as styles from './styles';
+import { useUpdateEffect } from '../utils/hooks';
+import Backdrop from './segmented-control-backdrop';
+import Button from './segmented-control-button';
+import type { SegmentedControlProps } from './types';
+
+const noop = () => {};
+
+function SegmentControl(
+	props: PolymorphicComponentProps< SegmentedControlProps, 'input' >,
+	forwardedRef: import('react').Ref< any >
+) {
+	const {
+		className,
+		baseId,
+		isAdaptiveWidth = false,
+		isBlock = false,
+		id,
+		label,
+		options = [],
+		onChange = noop,
+		value,
+		...otherProps
+	} = useContextSystem( props, 'SegmentedControl' );
+
+	const containerRef = useRef();
+	const [ resizeListener, sizes ] = useResizeAware();
+
+	const radio = useRadioState( {
+		baseId: baseId || id,
+		unstable_virtual: true,
```

Comment:
I think we should remove this line. This is what's producing the storybook warning (https://github.com/WordPress/gutenberg/pull/31937#discussion_r653333077). There will be a difference in how the focus is managed though (individual elements will receive focus instead of focusing on the container), so styles may need some adjustment.

```suggestion
		baseId: baseId || id,
```